pulpcore/cli/deb/content.py

```diff
@@ -0,0 +1,327 @@
+        entity_ctx.entity = {"sha256": value}
+    return value
+
+
+def _sha256_artifact_callback(
+    ctx: click.Context, param: click.Parameter, value: Optional[str]
+) -> Optional[Union[str, PulpEntityContext]]:
+    # Pass None and "" verbatim
+    if value:
+        pulp_ctx = ctx.find_object(PulpCLIContext)
+        assert pulp_ctx is not None
+        return PulpArtifactContext(pulp_ctx, entity={"sha256": value})
+    return value
+
+
+repository_option = resource_option(
+    "--repository",
+    default_plugin="deb",
+    default_type="deb",
+    context_table={
+        "deb:deb": PulpAptRepositoryContext,
+    },
+    href_pattern=PulpAptRepositoryContext.HREF_PATTERN,
+    help=_(
+        "Repository to add the content to in the form '[[<plugin>:]<resource_type>:]<name>' or by "
+        "href."
+    ),
+    allowed_with_contexts=(PulpDebPackageContext,),
+)
+
+
+@pulp_group()
+@type_option(
+    choices={
+        "generic_content": PulpDebGenericContentContext,
+        "installer_file_index": PulpDebInstallerFileIndexContext,
+        "installer_package": PulpDebInstallerPackageContext,
+        "package_release_component": PulpDebPackageReleaseComponentContext,
+        "package": PulpDebPackageContext,
+        "release_architecture": PulpDebReleaseArchitectureContext,
+        "release_component": PulpDebReleaseComponentContext,
+        "release_file": PulpDebReleaseFileContext,
+        "release": PulpDebReleaseContext,
+    },
+    default="package",
+    case_sensitive=False,
+)
+def content() -> None:
+    pass
+
+
+list_options = [
+    field_option,
+    exclude_field_option,
+    pulp_option("--repository-version"),
+    pulp_option(
+        "--architecture",
+        allowed_with_contexts=(
+            PulpDebInstallerFileIndexContext,
+            PulpDebInstallerPackageContext,
+            PulpDebPackageIndexContext,
+            PulpDebPackageContext,
+            PulpDebReleaseArchitectureContext,
+        ),
+    ),
+    pulp_option(
+        "--codename",
+        allowed_with_contexts=(
+            PulpDebReleaseArchitectureContext,
+            PulpDebReleaseComponentContext,
+            PulpDebReleaseFileContext,
+            PulpDebReleaseContext,
+        ),
+    ),
+    pulp_option(
+        "--distribution",
+        allowed_with_contexts=(
+            PulpDebReleaseArchitectureContext,
+            PulpDebReleaseComponentContext,
+            PulpDebReleaseContext,
+        ),
+    ),
+    pulp_option(
+        "--suite",
+        allowed_with_contexts=(
+            PulpDebReleaseArchitectureContext,
+            PulpDebReleaseComponentContext,
+            PulpDebReleaseFileContext,
+            PulpDebReleaseContext,
+        ),
+    ),
+    # TODO: IS THIS NEEDED?
+    pulp_option(
+        "--auto-built-package",
+        allowed_with_contexts=(
+            PulpDebInstallerPackageContext,
+            PulpDebPackageContext,
+        ),
+    ),
+    # TODO: IS THIS NEEDED?
```

Comment:
I  don't really know if this is the best way to implement it since different options work for different contexts however they all we be shown in the `--help` regardless. So it is possible to use an option for a context that doesn't do anything.

Second I don't really know which of the filter options are really needed I just added all I could find in the API docs so we may want to cut the ones we don't really want/use.

Reply:
I'd vote for keeping them, because eventually someone will need it for some reason :grin: